src/modules/commander/Arming/PreFlightCheck/checks/ekf2Check.cpp

```diff
@@ -79,16 +79,24 @@ bool PreFlightCheck::ekf2Check(orb_advert_t *mavlink_log_pub, vehicle_status_s &
 
 			} else if (status.pre_flt_fail_innov_height) {
 				mavlink_log_critical(mavlink_log_pub, "Preflight Fail: height estimate not stable");
-
-			} else if (status.pre_flt_fail_mag_field_disturbed) {
-				mavlink_log_critical(mavlink_log_pub, "Preflight Fail: strong magnetic interference detected");
 			}
 		}
 
 		success = false;
 		goto out;
 	}
 
+	param_get(param_find("COM_ARM_MAG_STR"), &is_mag_strength_check);
+
+	if (status.pre_flt_fail_mag_field_disturbed && (is_mag_strength_check == 1)) {
```

Comment:
```suggestion
	if ((mag_strength_check_enabled == 1) && status.pre_flt_fail_mag_field_disturbed) {
```
Is more readable IMHO.